documentation/modules/managing/con-producer-config-properties.adoc

```diff
@@ -0,0 +1,219 @@
+
+Broker restarts have a significant impact on high percentile statistics.
+For example, over a long period the 99th percentile latency is dominated by behavior around broker restarts.
+This is worth considering when designing benchmarks or comparing performance numbers from benchmarking with performance numbers seen in production.
+
+Depending on your objective, Kafka offers a number of configuration parameters and techniques for tuning producer performance for throughput and latency.
+
+Message batching (`linger.ms` and `batch.size`)::
+Message batching delays sending messages in the hope that more messages destined for the same broker will be sent,
+allowing them to be batched into a single produce request.
+Batching is a compromise between higher latency in return for higher throughput.
+Time-based batching is configured using `linger.ms`, and size-based batching is configured using `batch.size`.
+
+Compression (`compression.type`)::
+Message compression adds latency in the producer (CPU time spent compressing the messages),
+but makes requests (and potentially disk writes) smaller, which can increase throughput.
+Whether compression is worthwhile, and the best compression to use, will depend on the messages being sent.
+Compression happens on the thread which calls `KafkaProducer.send()`,
+so if the latency of this method matters for your application you should consider using more threads.
+
+Pipelining (`max.in.flight.requests.per.connection`)::
+Pipelining means sending more requests before the response to a previous request has been received.
+In general more pipelining means better throughput, up to a threshold at which other effects,
+such as worse batching, start to counteract the effect on throughput.
+
+.Lowering latency
+
+When your application calls `KafkaProducer.send()` the messages are:
+
+* Processed by any interceptors
+* Serialized
+* Assigned to a partition
+* Compressed
+* Added to a batch of messages in a per-partition queue
+
+At which point the `send()` method returns.
+So the time `send()` is blocked is determined by:
+
+* The time spent in the interceptors, serializers and partitioner
+* The compression algorithm used
+* The time spent waiting for a buffer to use for compression
+
+Batches will remain in the queue until one of the following occurs:
+
+* The batch is full (according to `batch.size`)
+* The delay introduced by `linger.ms` has passed
+* The sender is about to send message batches for other partitions to the same broker, and it is possible to add this batch too
+* The producer is being flushed or closed
+
+Look at the configuration for batching and buffering to mitigate the impact of `send()` blocking on latency.
+
+----
+# ...
+linger.ms=100 <1>
+batch.size=16384 <2>
+buffer.memory=33554432 <3>
+# ...
+----
+<1> The `linger` property adds a delay in milliseconds so that larger batches of messages are accumulated and sent in a request. The defaullt is `0'.`
```

Comment:
```suggestion
<1> The `linger` property adds a delay in milliseconds so that larger batches of messages are accumulated and sent in a request. The default is `0'.`
```